subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -334,17 +330,9 @@ public void visitOutputProperties(File workspace, OutputPropertyVisitor visitor)
                     visitor.visitOutputProperty(property.getPropertyName(), property.getOutputType(), outputFile, property.getPropertyFiles());
                 }
             }
-        }
-
-        @Override
-        public void visitLocalState(LocalStateVisitor visitor) {
             for (File localStateRoot : context.getTaskProperties().getLocalStateFiles()) {
```

Comment:
I wonder if destroyables and local state fit well in the `visitOutputProperties` method. Those are not really outputs, but locations the task modifies. Shall we try to come up with a better name for those things?

Reply:
Local state _is_ kind of an output, though not one that is shared, but I see how that's problematic. And destroyables are even less like an output property indeed.

How about we go for `visitInputs()` and `visitOutputs()` instead? In that case `visitInputs()` could take care of visiting the implementation(s), too.

Another option is to call them something like `visitConsumed()` and `visitProduced()` or `visitMutated()` but those seem to be too abstract/complicated.

Note that we only use local state and destroyables in tasks, and while local state might be something we add for other things, destroyables will always stay a task-specific thing.

Reply:
I think I'd go with `visitInputs()` and `visitOutputs()`, the latter handling local state, too, and perhaps adding back `visitDestroyed()` with a null implementation... WDYT?

(We could also capture a property name for local state BTW. But that's beyond this exercise.)

Reply:
We might just go with `visitInputs()` and `visitOutputs()`, and leave destroyables in `visitInputs()` too, and move it if it becomes an issue for someone.

I'd like to work towards reducing the API size of `UnitOfWork` as much as possible, and not having a specific `visitDestroyed()` with its accompanying visitor type sounds like a good direction for that.

Reply:
I am good with using `visitInputs` and `visitOutputs` as names.
I think `visitDestroyed()` needs to be part of `visitOutputs()`, since it is part of the things the work item changes, which determines how we can schedule the work.

Would it make sense to have default empty implementations for all methods in the visitors, so a caller of `visitInputs` and `visitOutputs` can then only override the methods they are interested in? I don't think having only a default implementation for `visitDestroyed()` makes sense.